crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -38,6 +41,263 @@ async fn local_node_price_improvement_fee_sell_order() {
+
+    tracing::info!("Rebalancing AMM pool.");
+    onchain
+        .mint_token_to_weth_uni_v2_pool(&limit_order_token, to_wei(1000))
+        .await;
+
+    tracing::info!("Waiting for liquidity state to update");
+    wait_for_condition(TIMEOUT, || async {
+        // Mint blocks until we evict the cached liquidity and fetch the new state.
+        onchain.mint_block().await;
+        let next = services.submit_quote(&quote_request).await.unwrap();
+        next.quote.buy_amount != quote.quote.buy_amount
+    })
+    .await
+    .unwrap();
+
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 3 })
+        .await
+        .unwrap();
+
+    wait_for_condition(TIMEOUT, || async { services.solvable_orders().await == 0 })
+        .await
+        .unwrap();
+
+    tracing::info!("Waiting for orders metadata update.");
+    let metadata_updated = || async {
+        onchain.mint_block().await;
+        let order_a = services
+            .get_order(&market_price_improvement_uid)
+            .await
+            .unwrap();
+        let order_b = services.get_order(&limit_surplus_order_uid).await.unwrap();
+        let order_c = services.get_order(&partner_fee_order_uid).await.unwrap();
+        !order_a.metadata.executed_surplus_fee.is_zero()
+            && !order_b.metadata.executed_surplus_fee.is_zero()
+            && !order_c.metadata.executed_surplus_fee.is_zero()
+    };
+    wait_for_condition(TIMEOUT, metadata_updated).await.unwrap();
+
+    let market_price_improvement = services
+        .get_order(&market_price_improvement_uid)
+        .await
+        .unwrap();
+    assert_approximately(
+        market_price_improvement.metadata.executed_surplus_fee,
+        202972993230628u128.into(),
+    );
+    let limit_surplus_order = services.get_order(&limit_surplus_order_uid).await.unwrap();
+    assert_approximately(
+        limit_surplus_order.metadata.executed_surplus_fee,
+        1473330174389045481u128.into(),
+    );
+    let partner_fee_order = services.get_order(&partner_fee_order_uid).await.unwrap();
+    assert_approximately(
+        partner_fee_order.metadata.executed_surplus_fee,
+        100163063434215497u128.into(),
+    );
+
+    let balance_after = market_order_token
+        .balance_of(onchain.contracts().gp_settlement.address())
+        .call()
+        .await
+        .unwrap();
+    assert_approximately(balance_after, 135044427261218u128.into());
+
+    let balance_after = limit_order_token
+        .balance_of(onchain.contracts().gp_settlement.address())
+        .call()
+        .await
+        .unwrap();
+    assert_approximately(balance_after, 99005624671640782648u128.into());
+
+    let balance_after = partner_fee_order_token
+        .balance_of(onchain.contracts().gp_settlement.address())
+        .call()
+        .await
+        .unwrap();
+    assert_approximately(balance_after, 66641691187129028u128.into());
```

Comment:
I think instead of using a magic number here (which we likely computed by running the test once and then blindly accepting its outcome as the right value to assert), I think a more robust test would be to assert that balance_after is > 2% of the sell amount.

This would give a more meaningful and robust assertion.